README.md

```diff
@@ -1,2 +1,233 @@
+   ii. Provide a name for the Service Provider (ex:- sample-app) and click `Register`. Now you will be redirected to the
+    `Edit Service Provider` page.<br/>
+   iii. Expand the  `Inbound Authentication Configuration` section and click `Configure` under the `OAuth/OpenID Connect Configuration` section.<br/>
+   iv. Provide the following values for the respective fields and click `Update` while keeping other default settings as it is.
+
+       Callback Url - regexp=(http://localhost:8080/callback/|http://localhost:8080/postlogout/)
+   v. Click `Update` to save.
+
+3. Once the service provider is saved, you will be redirected to the `Service Provider Details` page. Here, expand the
+    `Inbound Authentication Configuration` section and click the `OAuth/OpenID Connect Configuration` section. Copy the
+    values of  `OAuth Client Key` and `OAuth Client Secret` shown here.
+    ![OAuth Client Credentials](https://user-images.githubusercontent.com/15249242/91567068-27155e00-e962-11ea-8eab-b3bdd790bfd4.png)
+
+### Configuring the sample
+1. Download the sample application setup from [here](link).
+2. Double click the `setup.exe`.
+3. Follow the on-screen guidance until you get to the app configuration window.
+   ![forms](https://user-images.githubusercontent.com/15249242/95334388-a84e0200-08cb-11eb-89e6-e44d8ac158e4.gif)
+4. Fill out the following fields.
+
+       Client ID - <Enter the copied value of `OAuth Client Key` when creating the Service Provider>
+       Client Secret - <Enter the copied value of `OAuth Client Secret` when creating the Service Provider>
+       Authorization Endpoint - https://localhost:9443/oauth2/authorize
+       Token Endpoint - https://localhost:9443/oauth2/token
+       Userinfo Endpoint - https://localhost:9443/oauth2/userinfo
+       Logout Endpoint - https://localhost:9443/oidc/logout
+       Redirect URI - http://localhost:8080/callback/
+       PostLogout Redirect URI - http://localhost:8080/postlogout/
+5. Continue the on-screen guidance and complete the installation.
+
+### Running the sample
+Double click on the `Pickup Manager` application available on your Desktop.<br/>
+![pickup manager](https://user-images.githubusercontent.com/15249242/95334396-aa17c580-08cb-11eb-83ee-3b88b8512f68.gif)
+
+## How it works
+This section explains detailed walkthrough on how key aspects handled in the Asgardio .NET OIDC SDK.
+Througout this section we will refer to the source folder of the sample as <APP_HOME>
+
+  * [Trigger authentication](#trigger-authentication)
+  * [Retrieve user attributes](#retrieve-user-attributes)
+  * [Trigger logout](#trigger-logout)
+
+The structure of the sample would be as follows:<br/>
+![structure](https://user-images.githubusercontent.com/15249242/95333060-e34f3600-08c9-11eb-91e3-8ea64446eb77.PNG)
+
+### Trigger authentication
+In the `<APP_HOME>/LoginPage.xaml` page, we have registered a `Click` event named `LoginButton_Click` for the login button to trigger a OIDC authentication:
```

Comment:
to trigger an*

Reply:
Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006